stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -81,8 +84,10 @@
     @Nullable private Customer mCustomer;
     private long mCustomerCacheTime;
     @Nullable private Context mContext;
-    @Nullable private CustomerRetrievalListener mCustomerRetrievalListener;
-    @Nullable private SourceRetrievalListener mSourceRetrievalListener;
+    @NonNull private final Map<String, WeakReference<CustomerRetrievalListener>>
```

Comment:
Is there really a need for weakreferences here? I'm concerned it would bring back that issue of listeners getting garbage collected, and I don't see these leaking any contexts ass

Reply:
These listeners can have references to Activity contexts (see `PaymentMethodsActivity`), so they should be WeakReferences to avoid memory leaks. We'll monitor new reports of GC issues.